### PR TITLE
refactor(subscriptions): use account's settings data to check for active subscriptions

### LINKED ISSUE #1921: [content-server] Update Active Subscriptions Check for Subscription Management/Support to Use Account.settingsData
PR #1846 is adding `settingsData` to the Account model, which contains a list of subscriptions.  We could use that when checking for active subscriptions before displaying the Subscriptions subpanel or the support form.

### PR DESCRIPTION
**Pull request by chenba**
In the Support Form and the Subscriptions subpanel in Settings, we check
to see if the account has any active subscriptions.  Previously we were
fetching the subscriptions list directly from the active subscriptions
endpoint.  However, since a list of subscriptions is available in the
account model's settings data, we can make use of it, and save a few
requests.

Fixes #1921

@mozilla/fxa-devs r?

### CODE REVIEW

File: packages/fxa-content-server/app/tests/spec/models/account.js [hunk 1]
```diff
@@ -2972,13 +2972,45 @@ describe('models/account', function() {
           return Promise.resolve(subs);
         });
 
-      account.fetchActiveSubscriptions().then(resp => {
+      return account.fetchActiveSubscriptions().then(resp => {
```
Comment (chenba) [inline packages/fxa-content-server/app/tests/spec/models/account.js]: This change is not directly related to the issue/PR.  But this test was not running because it was missing the `return`.

File: packages/fxa-content-server/app/scripts/models/account.js [hunk 2]
```diff
@@ -1014,6 +1014,19 @@ const Account = Backbone.Model.extend(
       return this._fxaClient.attachedClients(this.get('sessionToken'));
     },
 
+    /**
+     * Check to see if the account has any active subscriptions.
+     *
+     * @returns {Promise} resolves to a bool.
+     */
+    hasActiveSubscriptions() {
```
Comment (lmorchard) [inline packages/fxa-content-server/app/scripts/models/account.js]: Nice little refactor, moving this to a utility method :+1:

File: packages/fxa-content-server/app/scripts/models/account.js [hunk 3]
```diff
@@ -1014,6 +1014,19 @@ const Account = Backbone.Model.extend(
...
+      return this.settingsData().then(
+        settingsData =>
+          Array.isArray(settingsData.subscriptions) &&
+          settingsData.subscriptions.some(s => s.status === 'active')
```
Comment (lmorchard) [inline packages/fxa-content-server/app/scripts/models/account.js]: I don't think we want this filter on the list - just a `settingsData.subscriptions.length > 0` should do.

I *think* [the `status` will equal `past_due` or `incomplete`](https://stripe.com/docs/api/subscriptions/object#subscription_object-status) in the case where the user's card has expired or been declined on an automatic payment. That's when [we display a warning on the settings page](https://github.com/mozilla/fxa/issues/970) and the user will need to click a link in the warning or the button to manage subscriptions and update their billing method. 

Technically, their subscription will be *not* active at that point, but I think we'll still want to offer access to the management pages.

File: packages/fxa-content-server/app/tests/spec/models/account.js (hunk 1 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/app/tests/spec/models/account.js]: great catch!

File: packages/fxa-content-server/app/scripts/models/account.js (hunk 3 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/app/scripts/models/account.js]: > Technically, their subscription will be not active at that point, but I think we'll still want to offer access to the management pages.

This seems reasonable so that users will be able to re-activate their subscription or update payment methods.

File: packages/fxa-content-server/app/scripts/views/settings/subscription.js [hunk 4]
```diff
@@ -30,22 +30,13 @@ const View = FormView.extend({
...
-   * @returns {Promise} resolves to a bool.
-   */
-  _hasActiveSubscriptions() {
     return this.getSignedInAccount()
-      .fetchActiveSubscriptions()
-      .then(subscriptions => subscriptions.length > 0);
+      .hasActiveSubscriptions()
+      .then(hasSubs => {
+        if (hasSubs) {
+          this.$el.html(this.renderTemplate(Template));
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/app/scripts/views/settings/subscription.js]: If there are no subscriptions, does this panel appear empty? What you can do is instead of calling render here and having template() return an empty string is use the fact that if `afterRender` returns false, the panel shouldn't render:

```js

template: Template,
...
afterRender() {
  return this.getSignedInAccount()
  .hasActiveSubscriptions()
  .then(hasSubs => {
    // view is not rendered if hasSubs is falsey.
    return !! hasSubs;
  });
}
```

I think that should work within settings panels too, it definitely works in normal views.

**Comment by chenba** [comment]
Pushed feedback based changes.  I've also renamed the method to `hasSubscriptions` to reflect that we don't check the `status` property for 'active'.  (Sounds a little awkward?  `isSubscriber`?)

File: packages/fxa-content-server/app/scripts/views/settings/subscription.js [hunk 5]
```diff
@@ -25,27 +26,8 @@ const View = FormView.extend({
...
-  /**
-   * Check to see if the account has any active subscriptions.
-   *
-   * @returns {Promise} resolves to a bool.
-   */
-  _hasActiveSubscriptions() {
-    return this.getSignedInAccount()
-      .fetchActiveSubscriptions()
-      .then(subscriptions => subscriptions.length > 0);
+    return this.getSignedInAccount().hasSubscriptions();
```
Comment (chenba) [inline packages/fxa-content-server/app/scripts/views/settings/subscription.js]: The rendering related changes here is from @shane-tomlinson [pointing out that we can control view rendering in `afterRender`](https://github.com/mozilla/fxa/pull/2079#discussion_r311106508).
